### PR TITLE
[JS API] Reschedule card with x days

### LINKED ISSUE #9727: How to reschedule a card via AnkiDroidJS?
Is there a way to reschedule a card via AnkiDroidJS? By reschedule I mean preforming the same action that is done when using the button "reschedule for review in x days".

**Is your feature request related to a problem? Please describe.**
I have written a script  in my card template that determine if a card has an interval above 30 days and if yes, will first show either the front or the back based on the evenness of the number of review for this card. 
If I see the front first, then I will dispose the card as normal (with one of the 4 the Anki buttons: again,...).
But if I see the back first, I want to add a HTML button to reschedule the card in X days (X been the number of days of the current interval). I don't want to use the Anki button so that the current interval and other settings of the card aren't updated by this review. 

Is this currently possible?

### PR DESCRIPTION
**Pull request by krmanik**
## Purpose / Description
Reschedule a card using JS API with x days.

## Fixes
Fixes #9727

## Approach
Used existing implemented function to reschedule the card.
```kotlin
RescheduleCards(cardIds, days).runWithHandler(scheduleCollectionTaskHandler(R.plurals.reschedule_cards_dialog_acknowledge))
```

## How Has This Been Tested?
Tested on devices and implemented Junit test.

## Learning (optional, can help others)
https://github.com/ankidroid/Anki-Android/wiki/AnkiDroid-Javascript-API

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidJsAPIConstants.kt [hunk 1]
```diff
@@ -29,6 +29,8 @@ object AnkiDroidJsAPIConstants {
     const val ankiJsErrorCodeBuryNote: Int = 5
     const val ankiJsErrorCodeSuspendNote: Int = 6
 
+    const val ankiJsErrorCodeSetDue: Int = 7
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidJsAPIConstants.kt:32]:
```suggestion
    const val ankiJsErrorCodeSetDue: Int = 7
```

File: AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidJsAPIConstants.kt [hunk 2]
```diff
@@ -41,6 +43,8 @@ object AnkiDroidJsAPIConstants {
     const val SUSPEND_CARD = "suspendCard"
     const val SUSPEND_NOTE = "suspendNote"
 
+    const val SET_CARD_DUE = "setCardDue"
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidJsAPIConstants.kt:46]:
```suggestion
    const val SET_CARD_DUE = "setCardDue"
```

File: AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidJsAPIConstants.kt [hunk 3]
```diff
@@ -51,6 +55,8 @@ object AnkiDroidJsAPIConstants {
         jsApiListMap[SUSPEND_CARD] = false
         jsApiListMap[SUSPEND_NOTE] = false
 
+        jsApiListMap[SET_CARD_DUE] = false
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidJsAPIConstants.kt:58]:
```suggestion
        jsApiListMap[SET_CARD_DUE] = false
```

**Review by mikehardy — APPROVED** [review]
tiniest of nitpicks, it just seemed like the spacing in the lists was inconsistent?
Otherwise looks good to me, I'm +1 / "set it to pending merge" if it gets another approval

File: AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidJsAPIConstants.kt (hunk 1 above)
Comment (krmanik) [reply AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidJsAPIConstants.kt]: Done, Thanks

**Review by david-allison — APPROVED** [review]

**Comment by silidev** [comment]
From the code of 2.17 alpha and 2.16: ([source](https://github.com/ankidroid/Anki-Android/blob/953b9bd879269910ee962b520da9705336d8bc2d/AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidJsAPIConstants.kt#L55)): 
```
        jsApiListMap[SET_CARD_DUE] = false
```

These flags prevent that the corresponding API call can be used. 
SET_CARD_DUE is **nowhere ** in the code this is set to true. My API init call also return it as false. Why is that so?
To me this issue looks like it should be enabled?

**Pull request merged**